Repo cleanup

andreasphil opened the pull request:
This PR performs various housekeeping in the repo with the goals of **reducing clutter, removing outdated information and files, and increasing clarity:**

- Updated the environment variable examples to include all relevant variables
- Removed the redundant top level `node-version` file: if I remember correctly this was needed due to a quirk in the setup of one developer, who is not in the team anymore. Please let me know if this causes issues for anyone.
- Merged German and English Codes of Conduct, to they can both be referenced at the same time and are consistent with our Contribution Guidelines.
- Removed the E2E test Docker setup, which was broken anyways and apparently not used by anyone in a long time
- Rename the `DockerfileApp` to a more standard `Dockerfile`

It also overhauls the READMEs, most importantly at the repo root, as well as the frontend and backend:

- Goals:
  - Quick reference for experienced Norms devies, who just want to look something up
  - Give people not familiar with the project an idea about how it works, without trying to be exhaustive
- Consider the system as a whole (frontend + backend + infra) instead of isolated modules
- Each piece of information should exist once and be referenced rather than duplicated (except for the quickstart)
- All outdated information removed

andreasphil committed:
fix: add missing env variables to examples

RISDEV-0000

andreasphil committed:
chore: delete top-level node-version file

This should not be needed as there is also a node-version specified for
the frontend.

RISDEV-0000

andreasphil committed:
docs: merge German and English codes of conduct

RISDEV-0000

andreasphil committed:
chore: remove E2E test Docker setup

It was broken anyways and has been for a while, so it's safe to say
nobody has been using it.

RISDEV-0000

andreasphil committed:
docs: rework README, merge DEVELOPING into README

RISDEV-0000

andreasphil committed:
docs: remove unnecessary note

RISDEV-0000

andreasphil committed:
docs: rework frontend and backend readme

RISDEV-0000

andreasphil committed:
chore: formatting

RISDEV-0000

andreasphil committed:
docs: move quickstart to the top

RISDEV-0000

andreasphil committed:
docs: phrasing, typos

RISDEV-0000

andreasphil committed:
refactor: rename DockerfileApp to Dockerfile

RISDEV-0000

malte-laukoetter commented on README.md:
```diff
@@ -1,53 +1,170 @@
+## Additional Resources
 
-See [`CONTRIBUTING.md`](./CONTRIBUTING.md) for details.
+- 🔒 [RIS Reports](https://github.com/digitalservicebund/ris-reports) (additional documentation, including architecture diagrams and JavaDocs)
+- 🔒 [Infrastructure](https://github.com/digitalservicebund/ris-norms-infra)
+- 🔒 [Importer for migrated data](https://github.com/digitalservicebund/ris-norms-migration-import/)
```
I'm not sure if we want to link them here as they are not public (or we should atleast add a comment about that so no one is confused)

malte-laukoetter replied:
oh that is what the lock is supposed to indicate, than don't mind my comment :)

andreasphil replied:
Exactly! It's a bit weird because we're partly public and partly not, but since most people working with this are DS people who do have access to the resources I wanted to include them

malte-laukoetter reviewed (approved):
Thanks a lot for cleaning up the readme files! It's a lot cleaner and simpler to understand now :)

VictorDelCampo commented on frontend/README.md:
```diff
@@ -1,104 +1,128 @@
+The frontend also includes our E2E tests.
 
-## Prerequisites
+Before diving into the code, please get familiar with our [code conventions](https://digitalservicebund.atlassian.net/wiki/x/BIC1N).
```
This should also have the "lock" symbol because not publicly available?

VictorDelCampo commented on monitoring/README.md:
```diff
@@ -1,3 +1,5 @@
+# Monitoring
```
Not really sure what do we have the `monitoring` module anymore ...

andreasphil replied:
Do we still need/use it? Otherwise I can remove it. I assumed that is how our Grafana dashboards are created?

VictorDelCampo replied:
Will discuss this topic with Sebastian. You can close your PR if this was the only open topic

VictorDelCampo commented on backend/README.md:
```diff
@@ -1,123 +1,91 @@
+# Backend
 
-Backend service built with Spring Boot.
+The Norms backend is a [Spring Boot](https://docs.spring.io/spring-boot/index.html) application build with Java. We use [Postgres](https://www.postgresql.org/) as our database and [Keycloak](https://www.keycloak.org/) as our identity provider. Both run locally using [Docker](https://www.docker.com/). In a live system (staging + production), the backend also serves the [frontend](../frontend/).
```
missing UAT? :) when you say staging + production, though don't think relevant at all to say staging and production

andreasphil replied:
Was just intended as examples, will update it

VictorDelCampo commented on backend/README.md:
````diff
@@ -1,123 +1,91 @@
+To run a check locally:
 
-```bash
-./gradlew checkLicense
````
You remove the part about the backend license? Why?

andreasphil replied:
I generally removed the docs about the license checker (for the frontend too) because I don't consider it a thing that people actively use but more of an automated check that runs in the pipeline. And I tried to not list every single possible command, only the most relevant for developing. Open to adding it back in if you disagree

VictorDelCampo replied:
Maybe just leave it there to create awareness about licensing? includes also the command that you would need to run locally if we face a problem in the pipeline. But don't have such a big strong opinion. you decide

andreasphil replied:
There is a note about the licenses in the main README, it just doesn't include the specific command. Leaving it as it is

VictorDelCampo commented on backend/README.md:
````diff
@@ -1,123 +1,91 @@
 ```
 
-## Check for outdated libraries
+## Check for Outdated Libraries
````
I think i saw this "format" several times, why Outdated Libraries with capital letters at the beginning?

andreasphil replied:
Good question, it's just a habit that I don't know when I picked it up but am 100% sure I have not invented. Had to look it up, it's [Associated Press style title case](https://en.wikipedia.org/wiki/Title_case#AP_Stylebook) (even though I don't know if I always use it correctly).

andreasphil replied:
Made the casing more consistent for all headings

VictorDelCampo commented on README.md:
```diff
@@ -1,53 +1,170 @@
-- Additionally, there is the general [RIS Documentation](https://digitalservicebund.github.io/ris-reports/), which shows the RIS Norms
-  - [architecture diagrams](https://digitalservicebund.github.io/ris-reports/docs/architecture/diagrams_list.html)
-  - backend [JavaDocs](https://digitalservicebund.github.io/ris-reports/docs/backend-code-documentation/norms-java.html)
+Opt in to CI posting notifications for failing jobs to a particular Slack channel by setting a repository secret with the name `SLACK_WEBHOOK_URL`, containing a url for [incoming webhooks](https://api.slack.com/messaging/webhooks).
```
Not sure about this part of slack in the global README and if at all in any readme :S

andreasphil replied:
It was there before (in a different place) so I kept it, but wouldn't mind deleting it either. What do you prefer?

VictorDelCampo replied:
I would remove it cause i dont really understand why it should be there. It we had the info about what channel to follow i could maybe understand it (for new devs) but not this info about setting it up, cause it is anyway done in the settings of the GH repo

VictorDelCampo reviewed (commented):
Amazing clean up! Minor comments

andreasphil committed:
docs: minor clarifications and spelling after feedback

RISDEV-0000

andreasphil committed:
ci: consistently name all pipeline jobs

RISDEV-0000

andreasphil committed:
docs: remove note about Slack notifications

It's just a repo configuration and doesn't provide much information for
users, removing it for now.

RISDEV-0000
